client/app/components/queries/QueryEditor/QueryEditorControls.jsx

```diff
@@ -153,6 +156,14 @@ EditorControl.propTypes = {
       onToggle: PropTypes.func,
     }),
   ]),
+  autoLimitCheckboxProps: PropTypes.oneOfType([
+    PropTypes.bool, // `false` to hide
+    PropTypes.shape({
```

Comment:
Can we use `PropTypes.shape(AutoLimitCheckbox.propTypes)` here to avoid duplication?